### PR TITLE
add monaco-typecheck tasks

### PR DESCRIPTION
**Pull request by jrieken**

### CODE REVIEW

File: build/gulpfile.editor.js [hunk 1]
```diff
@@ -235,3 +235,60 @@ function filterStream(testFunc) {
...
+function createTscCompileTask(watch) {
+	return () => {
+		const createReporter = require('./lib/reporter').createReporter;
+
+		return new Promise((resolve, reject) => {
+			const args = ['./node_modules/.bin/tsc', '-p', './src/tsconfig.monaco.json', '--noEmit'];
+			if (watch) {
+				args.push('-w');
+			}
+			const child = cp.spawn(`node`, args, {
```
Comment (joaomoreno) [inline build/gulpfile.editor.js]: Maybe use `cp.fork`, in order to get the same node instance?

File: build/gulpfile.editor.js [hunk 2]
```diff
@@ -235,3 +235,60 @@ function filterStream(testFunc) {
...
+				cwd: path.join(__dirname, '..'),
+				// stdio: [null, 'pipe', 'inherit']
+			});
+			let errors = [];
+			let reporter = createReporter();
+			let report;
+			let magic = /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g;
```
Comment (joaomoreno) [inline build/gulpfile.editor.js]: skillz

Comment (jrieken) [reply build/gulpfile.editor.js]: google skillz

File: build/gulpfile.editor.js (hunk 1 above)
Comment (jrieken) [reply build/gulpfile.editor.js]: Yeah, can give it a try

**Comment by jrieken** [comment]
Had to revert the `cp.fork` change because it wasn't working in the combined task (for an unknown reason...)

**Pull request merged**